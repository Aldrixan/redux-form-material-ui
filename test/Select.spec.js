import { render, unmountComponentAtNode } from "react-dom"

import React from "react"
import ReduxFormMaterialUISelect from "../src/Select"
import { act } from "react-dom/test-utils"
import noop from "lodash.noop"

describe("Select", () => {
    let container = null
    beforeEach(() => {
        container = document.createElement("div")
        document.body.appendChild(container)
    })

    afterEach(() => {
        unmountComponentAtNode(container)
        container.remove()
        container = null
    })

    test("renders a Select", () => {
        act(() => {
            render(
                <ReduxFormMaterialUISelect
                    input={{ onChange: noop, name: "mySelect", value: "Foo" }}
                    meta={{}}
                    helperText="FooHelper"
                />,
                container
            )
        })
        let element = container.querySelector('input[type="hidden"]')
        expect(element).toBeDefined()
        expect(element.getAttribute("value")).toEqual("Foo")
        expect(container.querySelector('div[aria-haspopup="listbox"]')).toBeDefined();
        expect(container.querySelector("p").textContent).toEqual("FooHelper")
    })

    test("renders a Select with no error", () => {
        act(() => {
            render(
                <ReduxFormMaterialUISelect
                    input={{ onChange: noop, name: "mySelect", value: "Foo" }}
                    meta={{ error: "FooError" }}
                />,
                container
            )
        })
        let element = container.querySelector("p")
        expect(element).toBeNull()
    })

    test("renders a Select with an error", () => {
        act(() => {
            render(
                <ReduxFormMaterialUISelect
                    input={{ onChange: noop, name: "mySelect", value: "Foo" }}
                    meta={{ error: "FooError", touched: true }}
                />,
                container
            )
        })
        let element = container.querySelector("p")
        expect(element.textContent).toEqual("FooError")
    })

    test("renders a Select with no warning", () => {
        act(() => {
            render(
                <ReduxFormMaterialUISelect
                    input={{ onChange: noop, name: "mySelect", value: "Foo" }}
                    meta={{ warning: "FooWarn" }}
                />,
                container
            )
        })
        let element = container.querySelector("p")
        expect(element).toBeNull()
    })

    test("renders a Select with an warning", () => {
        act(() => {
            render(
                <ReduxFormMaterialUISelect
                    input={{ onChange: noop, name: "mySelect", value: "Foo" }}
                    meta={{ warning: "FooWarn", touched: true }}
                />,
                container
            )
        })
        let element = container.querySelector("p")
        expect(element.textContent).toEqual("FooWarn")
    })

    test("renders a Select without defaultValue", () => {
        act(() => {
            render(
                <ReduxFormMaterialUISelect
                    input={{ onChange: noop, name: "mySelect", value: "Foo" }}
                    meta={{ warning: "FooWarn", touched: true }}
                    defaultValue={5}
                />,
                container
            )
        })
        let element = container.querySelector('input[type="hidden"]')
        expect(element.getAttribute("value")).toEqual("Foo")
    })

    it("renders a Select with className", () => {
        act(() => {
            render(
                <ReduxFormMaterialUISelect
                    input={{ onChange: noop, name: "mySelect", value: "Foo" }}
                    meta={{}}
                    className="myClassName"
                />,
                container
            )
        })
        let element = container.querySelector("div.myClassName")
        expect(element).toBeDefined()
    })

    it("renders a Select with multiple", () => {
        act(() => {
            render(
                <ReduxFormMaterialUISelect
                    multiple={true}
                    input={{ onChange: noop, name: "myText", value: ["Foo", "Bar"] }}
                    meta={{}}
                />,
                container
            )
        })
        let element = container.querySelector('input[type="hidden"]')
        expect(element.getAttribute("value")).toEqual("Foo,Bar")
    })

    it("renders a Select with error when touched and helper text", () => {
        act(() => {
            render(
                <ReduxFormMaterialUISelect
                    input={{ onChange: noop, name: "myText", value: "Foo" }}
                    meta={{ error: "FooError", touched: true }}
                    helperText="My Helper Text"
                />,
                container
            )
        })
        let element = container.querySelector("p")
        expect(element.textContent).toEqual("FooError")
    })
})
