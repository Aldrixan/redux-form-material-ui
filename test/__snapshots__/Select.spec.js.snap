// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Select renders a Select 1`] = `
<div
  className="MuiFormControl-root"
>
  <label
    className="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-shrink MuiFormLabel-filled"
    data-shrink={true}
    htmlFor="myText"
  />
  <div
    className="MuiInputBase-root MuiInput-root MuiInput-underline MuiInputBase-formControl MuiInput-formControl"
    onClick={[Function]}
  >
    <div
      aria-haspopup="listbox"
      className="MuiSelect-root MuiSelect-select MuiSelect-selectMenu MuiInputBase-input MuiInput-input MuiInputBase-inputSelect"
      onBlur={[Function]}
      onClick={[Function]}
      onFocus={[Function]}
      onKeyDown={[Function]}
      role="button"
      tabIndex={0}
    >
      <span
        dangerouslySetInnerHTML={
          Object {
            "__html": "&#8203;",
          }
        }
      />
    </div>
    <input
      onAnimationStart={[Function]}
      type="hidden"
      value="Foo"
    />
    <svg
      aria-hidden="true"
      className="MuiSvgIcon-root MuiSelect-icon"
      focusable="false"
      role="presentation"
      viewBox="0 0 24 24"
    >
      <path
        d="M7 10l5 5 5-5z"
      />
    </svg>
  </div>
</div>
`;

exports[`Select renders a Select with a warning 1`] = `
<div
  className="MuiFormControl-root"
>
  <label
    className="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-shrink MuiFormLabel-filled"
    data-shrink={true}
    htmlFor="myText"
  />
  <div
    className="MuiInputBase-root MuiInput-root MuiInput-underline MuiInputBase-formControl MuiInput-formControl"
    onClick={[Function]}
  >
    <div
      aria-haspopup="listbox"
      className="MuiSelect-root MuiSelect-select MuiSelect-selectMenu MuiInputBase-input MuiInput-input MuiInputBase-inputSelect"
      onBlur={[Function]}
      onClick={[Function]}
      onFocus={[Function]}
      onKeyDown={[Function]}
      role="button"
      tabIndex={0}
    >
      <span
        dangerouslySetInnerHTML={
          Object {
            "__html": "&#8203;",
          }
        }
      />
    </div>
    <input
      onAnimationStart={[Function]}
      type="hidden"
      value="Foo"
    />
    <svg
      aria-hidden="true"
      className="MuiSvgIcon-root MuiSelect-icon"
      focusable="false"
      role="presentation"
      viewBox="0 0 24 24"
    >
      <path
        d="M7 10l5 5 5-5z"
      />
    </svg>
  </div>
  <p
    className="MuiFormHelperText-root MuiFormHelperText-filled"
  >
    FooWarning
  </p>
</div>
`;

exports[`Select renders a Select with an error 1`] = `
<div
  className="MuiFormControl-root"
>
  <label
    className="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-shrink Mui-error Mui-error MuiFormLabel-filled"
    data-shrink={true}
    htmlFor="myText"
  />
  <div
    className="MuiInputBase-root MuiInput-root MuiInput-underline Mui-error Mui-error MuiInputBase-formControl MuiInput-formControl"
    onClick={[Function]}
  >
    <div
      aria-haspopup="listbox"
      className="MuiSelect-root MuiSelect-select MuiSelect-selectMenu MuiInputBase-input MuiInput-input MuiInputBase-inputSelect"
      onBlur={[Function]}
      onClick={[Function]}
      onFocus={[Function]}
      onKeyDown={[Function]}
      role="button"
      tabIndex={0}
    >
      <span
        dangerouslySetInnerHTML={
          Object {
            "__html": "&#8203;",
          }
        }
      />
    </div>
    <input
      onAnimationStart={[Function]}
      type="hidden"
      value="Foo"
    />
    <svg
      aria-hidden="true"
      className="MuiSvgIcon-root MuiSelect-icon"
      focusable="false"
      role="presentation"
      viewBox="0 0 24 24"
    >
      <path
        d="M7 10l5 5 5-5z"
      />
    </svg>
  </div>
  <p
    className="MuiFormHelperText-root Mui-error MuiFormHelperText-filled"
  >
    FooError
  </p>
</div>
`;

exports[`Select renders a Select with className 1`] = `
<div
  className="MuiFormControl-root myClassName"
>
  <label
    className="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-shrink MuiFormLabel-filled"
    data-shrink={true}
    htmlFor="myText"
  />
  <div
    className="MuiInputBase-root MuiInput-root MuiInput-underline MuiInputBase-formControl MuiInput-formControl"
    onClick={[Function]}
  >
    <div
      aria-haspopup="listbox"
      className="MuiSelect-root MuiSelect-select MuiSelect-selectMenu MuiInputBase-input MuiInput-input MuiInputBase-inputSelect"
      onBlur={[Function]}
      onClick={[Function]}
      onFocus={[Function]}
      onKeyDown={[Function]}
      role="button"
      tabIndex={0}
    >
      <span
        dangerouslySetInnerHTML={
          Object {
            "__html": "&#8203;",
          }
        }
      />
    </div>
    <input
      onAnimationStart={[Function]}
      type="hidden"
      value="Foo"
    />
    <svg
      aria-hidden="true"
      className="MuiSvgIcon-root MuiSelect-icon"
      focusable="false"
      role="presentation"
      viewBox="0 0 24 24"
    >
      <path
        d="M7 10l5 5 5-5z"
      />
    </svg>
  </div>
</div>
`;

exports[`Select renders a Select with error when touched and helper text 1`] = `
<div
  className="MuiFormControl-root"
>
  <label
    className="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-shrink Mui-error Mui-error MuiFormLabel-filled"
    data-shrink={true}
    htmlFor="myText"
  />
  <div
    className="MuiInputBase-root MuiInput-root MuiInput-underline Mui-error Mui-error MuiInputBase-formControl MuiInput-formControl"
    onClick={[Function]}
  >
    <div
      aria-haspopup="listbox"
      className="MuiSelect-root MuiSelect-select MuiSelect-selectMenu MuiInputBase-input MuiInput-input MuiInputBase-inputSelect"
      onBlur={[Function]}
      onClick={[Function]}
      onFocus={[Function]}
      onKeyDown={[Function]}
      role="button"
      tabIndex={0}
    >
      <span
        dangerouslySetInnerHTML={
          Object {
            "__html": "&#8203;",
          }
        }
      />
    </div>
    <input
      onAnimationStart={[Function]}
      type="hidden"
      value="Foo"
    />
    <svg
      aria-hidden="true"
      className="MuiSvgIcon-root MuiSelect-icon"
      focusable="false"
      role="presentation"
      viewBox="0 0 24 24"
    >
      <path
        d="M7 10l5 5 5-5z"
      />
    </svg>
  </div>
  <p
    className="MuiFormHelperText-root Mui-error MuiFormHelperText-filled"
  >
    FooError
  </p>
</div>
`;

exports[`Select renders a Select with helper text when not touched 1`] = `
<div
  className="MuiFormControl-root"
>
  <label
    className="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-shrink MuiFormLabel-filled"
    data-shrink={true}
    htmlFor="myText"
  />
  <div
    className="MuiInputBase-root MuiInput-root MuiInput-underline MuiInputBase-formControl MuiInput-formControl"
    onClick={[Function]}
  >
    <div
      aria-haspopup="listbox"
      className="MuiSelect-root MuiSelect-select MuiSelect-selectMenu MuiInputBase-input MuiInput-input MuiInputBase-inputSelect"
      onBlur={[Function]}
      onClick={[Function]}
      onFocus={[Function]}
      onKeyDown={[Function]}
      role="button"
      tabIndex={0}
    >
      <span
        dangerouslySetInnerHTML={
          Object {
            "__html": "&#8203;",
          }
        }
      />
    </div>
    <input
      onAnimationStart={[Function]}
      type="hidden"
      value="Foo"
    />
    <svg
      aria-hidden="true"
      className="MuiSvgIcon-root MuiSelect-icon"
      focusable="false"
      role="presentation"
      viewBox="0 0 24 24"
    >
      <path
        d="M7 10l5 5 5-5z"
      />
    </svg>
  </div>
  <p
    className="MuiFormHelperText-root MuiFormHelperText-filled"
  >
    My Helper Text
  </p>
</div>
`;

exports[`Select renders a Select with multiple 1`] = `
<div
  className="MuiFormControl-root"
>
  <label
    className="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-shrink MuiFormLabel-filled"
    data-shrink={true}
    htmlFor="myText"
  />
  <div
    className="MuiInputBase-root MuiInput-root MuiInput-underline MuiInputBase-formControl MuiInput-formControl"
    onClick={[Function]}
  >
    <div
      aria-haspopup="listbox"
      className="MuiSelect-root MuiSelect-select MuiSelect-selectMenu MuiInputBase-input MuiInput-input MuiInputBase-inputSelect"
      onBlur={[Function]}
      onClick={[Function]}
      onFocus={[Function]}
      onKeyDown={[Function]}
      role="button"
      tabIndex={0}
    >
      <span
        dangerouslySetInnerHTML={
          Object {
            "__html": "&#8203;",
          }
        }
      />
    </div>
    <input
      onAnimationStart={[Function]}
      type="hidden"
      value="Foo,Bar"
    />
    <svg
      aria-hidden="true"
      className="MuiSvgIcon-root MuiSelect-icon"
      focusable="false"
      role="presentation"
      viewBox="0 0 24 24"
    >
      <path
        d="M7 10l5 5 5-5z"
      />
    </svg>
  </div>
</div>
`;

exports[`Select renders a Select with no error when not touched 1`] = `
<div
  className="MuiFormControl-root"
>
  <label
    className="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-shrink MuiFormLabel-filled"
    data-shrink={true}
    htmlFor="myText"
  />
  <div
    className="MuiInputBase-root MuiInput-root MuiInput-underline MuiInputBase-formControl MuiInput-formControl"
    onClick={[Function]}
  >
    <div
      aria-haspopup="listbox"
      className="MuiSelect-root MuiSelect-select MuiSelect-selectMenu MuiInputBase-input MuiInput-input MuiInputBase-inputSelect"
      onBlur={[Function]}
      onClick={[Function]}
      onFocus={[Function]}
      onKeyDown={[Function]}
      role="button"
      tabIndex={0}
    >
      <span
        dangerouslySetInnerHTML={
          Object {
            "__html": "&#8203;",
          }
        }
      />
    </div>
    <input
      onAnimationStart={[Function]}
      type="hidden"
      value="Foo"
    />
    <svg
      aria-hidden="true"
      className="MuiSvgIcon-root MuiSelect-icon"
      focusable="false"
      role="presentation"
      viewBox="0 0 24 24"
    >
      <path
        d="M7 10l5 5 5-5z"
      />
    </svg>
  </div>
</div>
`;

exports[`Select renders a Select with no warning when not touched 1`] = `
<div
  className="MuiFormControl-root"
>
  <label
    className="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-shrink MuiFormLabel-filled"
    data-shrink={true}
    htmlFor="myText"
  />
  <div
    className="MuiInputBase-root MuiInput-root MuiInput-underline MuiInputBase-formControl MuiInput-formControl"
    onClick={[Function]}
  >
    <div
      aria-haspopup="listbox"
      className="MuiSelect-root MuiSelect-select MuiSelect-selectMenu MuiInputBase-input MuiInput-input MuiInputBase-inputSelect"
      onBlur={[Function]}
      onClick={[Function]}
      onFocus={[Function]}
      onKeyDown={[Function]}
      role="button"
      tabIndex={0}
    >
      <span
        dangerouslySetInnerHTML={
          Object {
            "__html": "&#8203;",
          }
        }
      />
    </div>
    <input
      onAnimationStart={[Function]}
      type="hidden"
      value="Foo"
    />
    <svg
      aria-hidden="true"
      className="MuiSvgIcon-root MuiSelect-icon"
      focusable="false"
      role="presentation"
      viewBox="0 0 24 24"
    >
      <path
        d="M7 10l5 5 5-5z"
      />
    </svg>
  </div>
</div>
`;

exports[`Select should ignore defaultValue 1`] = `
<div
  className="MuiFormControl-root"
>
  <label
    className="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-shrink MuiFormLabel-filled"
    data-shrink={true}
    htmlFor="myText"
  />
  <div
    className="MuiInputBase-root MuiInput-root MuiInput-underline MuiInputBase-formControl MuiInput-formControl"
    onClick={[Function]}
  >
    <div
      aria-haspopup="listbox"
      className="MuiSelect-root MuiSelect-select MuiSelect-selectMenu MuiInputBase-input MuiInput-input MuiInputBase-inputSelect"
      onBlur={[Function]}
      onClick={[Function]}
      onFocus={[Function]}
      onKeyDown={[Function]}
      role="button"
      tabIndex={0}
    >
      <span
        dangerouslySetInnerHTML={
          Object {
            "__html": "&#8203;",
          }
        }
      />
    </div>
    <input
      onAnimationStart={[Function]}
      type="hidden"
      value="Foo"
    />
    <svg
      aria-hidden="true"
      className="MuiSvgIcon-root MuiSelect-icon"
      focusable="false"
      role="presentation"
      viewBox="0 0 24 24"
    >
      <path
        d="M7 10l5 5 5-5z"
      />
    </svg>
  </div>
</div>
`;
